### PR TITLE
Add QSD blockdev option testing

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:2124451
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=qsd_blockdev_check --guestname=RHEL.8.4.0 --clone=no --firmware=default_bios

(1/1) Host_RHEL.m8.u8.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qsd_blockdev_check.q35: STARTED
 (1/1) Host_RHEL.m8.u8.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qsd_blockdev_check.q35: PASS (8.44 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by qingwangrh** [comment]
@YongxueHong @zhenyzha Please help to review,thanks.

File: qemu/tests/cfg/qsd_blockdev_check.cfg [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    image_name_stg1 = images/stg1
+    image_size_stg1 = 128M
+    image_name_stg2 = images/stg2
+    image_size_stg2 = 256M
+
+    qsd_cmd_lines += " --object iothread,id=iothread1;"
+    # Image throttle and export attributes
+
+    qsd_image_protocol_stg1 = {"aio":"native","cache":{"direct":true,"no-flush":false}}
+    qsd_image_format_stg1 = {"driver":"qcow2","read-only":true,"detect-zeroes":"unmap","discard":"unmap","force-share":true}
```
Comment (YongxueHong) [inline qemu/tests/cfg/qsd_blockdev_check.cfg:19]: Suggest splitting it into multiple lines.

File: qemu/tests/cfg/qsd_blockdev_check.cfg [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    qsd_image_protocol_stg2 = {"aio":"threads","cache":{"direct":false,"no-flush":true}}
+    qsd_image_format_stg2 = {"driver":"raw","read-only":false,"detect-zeroes":"on","discard":"ignore","force-share":false}
```
Comment (YongxueHong) [inline qemu/tests/cfg/qsd_blockdev_check.cfg:21]: Suggest splitting it into multiple lines.

File: qemu/tests/qsd_blockdev_check.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            if node["node-name"] == "prot_%s" % img:
+                prot_node = node
+            if node["node-name"] == "fmt_%s" % img:
+                fmt_node = node
+        if not fmt_node or not prot_node:
+            test.fail("Can not find blockdev node")
+        nodes_data = [(prot_attrs, prot_node), (fmt_attrs, fmt_node)]
+        for check_data in nodes_data:
+            attrs = check_data[0]
+            node = check_data[1]
```
Comment (YongxueHong) [inline qemu/tests/qsd_blockdev_check.py:33]:
```suggestion
        for attrs, node in zip((prot_attrs, fmt_node), (prot_node, fmt_node)):
```

Comment (qingwangrh) [reply qemu/tests/qsd_blockdev_check.py]: updated

File: qemu/tests/cfg/qsd_blockdev_check.cfg (hunk 2 above)
Comment (qingwangrh) [reply qemu/tests/cfg/qsd_blockdev_check.cfg]: updated

**Comment by zhenyzha** [comment]
git am 3484.patch
Applying: Add QSD blockdev option testing

` (1/1) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.qsd_blockdev_check.arm64-pci: PASS (5.17 s)`

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

**Comment by XueqiangWei** [comment]
@YongxueHong Please help review it. Many thanks.

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**